Добрый день.

Kederly84 opened the pull request:
Выполнены все задания, в т.ч. и со звездочкой.
В задании 6_1 так же выполнено задание 6_2
Для задания 6_5 функцию вынес в отдельный файл utils.py
Для заданий 6_6 и 6_7 создал отдельную директорию, чтобы в основной не было мешанины.

Kederly84 committed:
Добрый день.
Выполнены все задания, в т.ч. и со звездочкой.
В задании 6_1 так же выполнено задание 6_2
Для задания 6_5 функцию вынес в отдельный файл utils.py
Для заданий 6_6 и 6_7 создал отдельную директорию, чтобы в основной не было мешанины.

shishkinav commented on Lesson_6/Aleksandr_Maksimov_Task_6_1.py:
```diff
@@ -0,0 +1,36 @@
+# Импорт стандартной библиотеки Python чтобы не сплитовать по несколько раз
+import re
```
торопитесь, говорил же в конце прошлого вебинара, не нужно сейчас пользовать регулярку

shishkinav commented on Lesson_6/Aleksandr_Maksimov_Task_6_3.py:
```diff
@@ -0,0 +1,31 @@
+# Импорт необходимых библиотек
+import sys
+
+def wording(file_1, file_2):
```
по стилистике двойная пустая строка должна быть между функцией и импортами

shishkinav commented on Lesson_6/Aleksandr_Maksimov_Task_6_5.py:
```diff
@@ -0,0 +1,16 @@
+
+def util(argv):
+    """Функция, которая передает необходимые аргументы в исполняемый файл"""
+    arg_1 = argv[-3]
+    arg_2 = argv[-2]
+    arg_3 = argv[-1]
```
было бы неплохо запилить проверку наличия архументов, потому что стоит не передать аргументы, вылетим в ошибку

shishkinav commented on Lesson_6/Aleksandr_Maksimov_Task_6_6/add_sale.py:
```diff
@@ -0,0 +1,22 @@
+# Запись данныех в файл для задачи 6_6
+
+def write(argv):
```
чтобы не было путаницы, лучше не называть функции именами, которые часто используются в методах разных объектов

shishkinav commented on Lesson_6/Aleksandr_Maksimov_Task_6_6/add_sale.py:
```diff
@@ -0,0 +1,22 @@
+        print('Задайте число которое необходимо сохранить')
+    else:
+        with open('Storage/Storage.csv', 'a+', encoding='utf-8') as f:
+            str_num = 1
+            f.seek(0)
+            line = f.readline()
+            while line:
+                str_num += 1
+                line = f.readline()
```
можно было перебор построить через
```
for idx, line in enumerate(f, start=1):
    <что-то делаете>
else:
    <действия когда всё перебрали>
```

shishkinav commented on Lesson_6/Aleksandr_Maksimov_Task_6_6/rewrite_sales.py:
```diff
@@ -0,0 +1,25 @@
+    num = argv[-1]
+    finish_list = []
+    with open('Storage/Storage.csv', 'r', encoding='utf-8') as f:
+        line = f.readline()
+        while line:
+            finish_list.append(line)
+            line = f.readline()
+    if check > len(finish_list):
```
зачем вычитывать построчно через `readline` если в конечном итоге вы всё складываете в `finish_list`? 
можно было эти четыре строки заменить  на одну:
```
finish_list = f.readlines()
```

shishkinav commented on Lesson_6/utils.py:
```diff
@@ -0,0 +1,26 @@
+        for row_2 in f_2:
+            hobby.append(row_2[:-1])
+    if len(usr) < len(hobby):
+        sys.exit(1)
+    def hobbies(n):
+        if n < len(hobby):
+            return hobby[n]
+        else:
+            return
+    for i in range(len(usr)):
```
даже внутри другой функции, когда пишите дополнительную локальную функцию - отделяйте её пустыми строками, она не должна сливаться со всем телом кода

shishkinav commented on Lesson_6/Aleksandr_Maksimov_Task_6_6/show_sales.py:
```diff
@@ -0,0 +1,59 @@
+                    pos_start = f.tell() - line_len
+                    break
+                line = f.readline()
+        if pos_start == None:
```
если требуется проверка pos_start, что он имеет значение None, то правильнее было бы написать `if pos_start is None:`

Kederly84 committed:
Добрый день.
Выполнены все задания, в т.ч. и со звездочкой.
В задании 6_1 так же выполнено задание 6_2
Для задания 6_5 функцию вынес в отдельный файл utils.py
Для заданий 6_6 и 6_7 создал отдельную директорию, чтобы в основной не было мешанины.
